authentication/utils.py

```diff
@@ -0,0 +1,26 @@
+from firebase_admin import auth
+from rest_framework.status import HTTP_422_UNPROCESSABLE_ENTITY, HTTP_400_BAD_REQUEST
+from rest_framework.exceptions import ValidationError
+
+class FirebaseAPI:
+    @classmethod
+    def verify_id_token(cls, id_token):
+        try:
+            decoded_token = auth.verify_id_token(id_token)
+            return decoded_token
+        except ValueError:
```

Comment:
```suggestion
        except:
```

Reply:
In the try-except block, I think it's a good practice to catch the proper exception, rather than catching all the exceptions. Firebase may raise 5 different types of exceptions here as mentioned in the [docs](https://firebase.google.com/docs/reference/admin/python/firebase_admin.auth#verify_id_token). Out of the 5, we may exclude `RevokedIdTokenError` because `check_revoked` is set to `False` by default.

So, either we should catch all the exceptions directly using `except` (as suggested by your change), or catch each of these 4 exceptions and raise the `ValidationError` with the proper message. Which one of the two would you prefer? @nishantwrp

Reply:
Not opposed to any solution. Both seem good!